gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngineTest.java

```diff
@@ -102,34 +102,43 @@ public boolean hasNext() {
     }
 
     @Override
-    public synchronized DagTask next() {
+    public synchronized DagTask next() throws NoSuchElementException {
       i++;
-      if (i <= MAX_NUM_OF_TASKS) {
-        return new MockedDagTask(new DagActionStore.DagAction("fg-" + i, "fn-" + i, "1234" + i, DagActionStore.FlowActionType.LAUNCH), null);
+      if (i > MAX_NUM_OF_TASKS) {
+        throw new RuntimeException("Simulating an exception to stop the thread!");
```

Comment:
technically this should be thrown by `MockedDagProc.act`... so a param to it's ctor (tunneled through the `MockedDagTask::host`)

Reply:
if the test throws exception in MockedDagProc.act , thread will not stop.... i just wanted to stop the DPE thread, though it is not required...
DagProcEngineThread is written in such a way that it will swallow exceptions (with logging) thrown by dag procs, but will die on any exception in dagTaskStream.next()